lib/managers/folders.js

```diff
@@ -301,6 +301,35 @@ Folders.prototype.deleteMetadata = function(folderID, scope, template, callback)
 	this.client.del(apiPath, null, this.client.defaultResponseHandler(callback));
 };
 
+/**
+ * Restores an item that has been moved to the trash. Default behavior is to restore the item
+ * to the folder it was in before it was moved to the trash. If that parent folder no longer
+ * exists or if there is now an item with the same name in that parent folder, the new parent
+ * older and/or new name will need to be included in the request.
+ *
+ * API Endpoint: '/folders/:folderID'
+ * Method: POST
+ *
+ * @param {string} folderID - The ID of the folder to restore
+ * @param {string} name - The new name for the folder
+ * @param {string} parentFolderID - The new parent folder for this item
+ * @param {Function} callback - Called with nothing if successful, error otherwise
+ * @returns {void}
+ */
+Folders.prototype.restoreFolder = function(folderID, name, parentFolderID, callback) {
+
+	var params = {
+		body: {
+			name: name,
+			parent: {
+				id: parentFolderID
+			}
```

Comment:
With the optional params, we'll need to make sure we only add them to the request body if they were passed into this function.